Source/Events.Handling/Builder/ConventionEventHandlerBuilder.cs

```diff
@@ -0,0 +1,319 @@
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConventionEventHandlerBuilder"/> class.
+        /// </summary>
+        /// <param name="eventHandlerType">The event handler <see cref="Type" />.</param>
+        public ConventionEventHandlerBuilder(Type eventHandlerType) => _eventHandlerType = eventHandlerType;
+
+        /// <inheritdoc/>
+        public BuildEventHandlerResult BuildAndRegister(
+            IEventProcessors eventProcessors,
+            IEventTypes eventTypes,
+            IEventProcessingConverter processingConverter,
+            IContainer container,
+            ILoggerFactory loggerFactory,
+            CancellationToken cancellation)
+        {
+            _eventHandlerInstance ??= container.Get(_eventHandlerType);
+            var logger = loggerFactory.CreateLogger<ConventionEventHandlerBuilder>();
+            logger.LogTrace("Building event handler from type {EventHandler}", _eventHandlerType);
+            if (!TryGetEventHandlerInformation(out var eventHandlerId, out var partitioned, out var scopeId))
+            {
+                return new BuildEventHandlerResult(
+                    null,
+                    $"The event handler class {_eventHandlerType} needs to be decorated with an [{typeof(EventHandlerAttribute).Name}]");
+            }
+
+            logger.LogTrace(
+                "Building {PartitionedOrUnpartitioned} event handler {EventHandlerId} processing events in scope {ScopeId} from type {EventHandler}",
+                partitioned ? "partitioned" : "unpartitioned",
+                eventHandlerId,
+                scopeId,
+                _eventHandlerType);
+
+            if (!TryBuildHandlerMethods(
+                eventHandlerId,
+                eventTypes,
+                out var eventTypesToMethods,
+                out var buildResult))
+            {
+                return buildResult;
+            }
+
+            var eventHandler = new EventHandler(eventHandlerId, scopeId, partitioned, eventTypesToMethods);
+            var eventHandlerProcessor = new EventHandlerProcessor(
+                eventHandler,
+                processingConverter,
+                loggerFactory.CreateLogger<EventHandlerProcessor>());
+            eventProcessors.Register(
+                eventHandlerProcessor,
+                new EventHandlerProtocol(),
+                cancellation);
+            return new BuildEventHandlerResult();
+        }
+
+        bool TryBuildHandlerMethods(
+            EventHandlerId eventHandlerId,
+            IEventTypes eventTypes,
+            out IDictionary<EventType, IEventHandlerMethod> eventTypesToMethods,
+            out BuildEventHandlerResult buildResult)
+        {
+            var eventHandlerMethodsBuilder = new EventHandlerMethodsBuilder(eventHandlerId);
+            var publicMethods = _eventHandlerType.GetMethods(BindingFlags.Instance | BindingFlags.DeclaredOnly | BindingFlags.Public);
```

Comment:
I think we used to have warnings if something looked like a Handler method, so had the Handle name or [Handles(..)] attribute now, but is non-public?

Reply:
I don't know if we should do that?